client/lib/jetpack/wpcom-upsell-controller.tsx

```diff
@@ -23,27 +21,24 @@ export default function upsellSwitch(
 		const siteId = getSelectedSiteId( getState() );
 		const isJetpack = isJetpackSite( getState(), siteId );
 
+		// Note that isJetpack *includes* Atomic sites, so we don't need to explicitly check for that.
 		if ( isJetpack || isEnabled( 'jetpack-cloud' ) ) {
 			return next();
 		}
 
-		const currentPlan = getSitePlan( getState(), siteId );
-		const isBusiness = isBusinessPlan( currentPlan.product_slug );
-
 		// Show Business plan upsell if it isn't a Jetpack site and it doesn't have
-		// a Business plan
-		if ( ! isBusiness ) {
+		// a Business plan. Note this will include E-Comm.
+		const hasPlanForAT = siteId && isSiteOnAtomicPlan( getState(), siteId );
+		// eslint-disable-next-line
+		if ( ! hasPlanForAT && false ) {
```

Comment:
So, the `false` is here as I am getting false negatives for `isSiteOnAtomicPlan`. The issue here is this type of function can't handle React loading states.